### ISSUE TITLE
Incorrect indentation with chained calls

### ISSUE THREAD

**Issue by berarma** [2072/PHP-Indenting-for-VIm#67]
I've found some cases where the arrow matching is suboptimal.

```
$article->author = $this->Authors->find()
        ->first();
```

Here, it should start matching after the assignment. Like this:

```
$article->author = $this->Authors->find()
                        ->first();
```

That's how it would work if there was no arrow before the assignment. Another improvement would be that the arrow matched the arrow of the first method call in the line. Like this:

```
$article->author = $this->Authors->find()
                                 ->first();
```

Thanks for the plugin.
Best regards.

**Comment by berarma** [comment]
Great! I can't think of any problem. I was just cautious to not propose a change that could break other cases but none that I know of.

Thanks!

**Comment by StanAngeloff** [comment]
For me this patch fails rather badly on the following code:

```php
Assert::that($vatRate->equals($item->getVatRate()))
    ->true('The VAT rates differ');
```

The `->true(…)` part should be indented relative to `Assert::that` not the expression within the parentheses.

**Comment by chronon** [comment]
For some projects I work on the coding standard requires a normal indent (whether tab or spaces), so I end up fighting where this fix puts things.

``` php
$article->author = $this->Authors->find()
    ->first()
    ->second();
```

**Comment by StanAngeloff** [comment]
@2072 perhaps a revert of 61ad820386ae905437a5187b7a098210f16f6832 is in order?

**Comment by berarma** [comment]
I wouldn't mind since I've disabled this feature. But be warned, it won't work always as expected with or without the patch.

**Comment by chronon** [comment]
Changing the default setting for `PHP_noArrowMatching` fixes this for me, I should have read the help more carefully. @StanAngeloff, possibly this will fix your use case as well?
```
let g:PHP_noArrowMatching = 1
```

**Comment by StanAngeloff** [comment]
> Changing the default setting for `PHP_noArrowMatching` fixes this for me, I should have read the help more carefully. @StanAngeloff, possibly this will fix your use case as well?
> 
> ```
> let g:PHP_noArrowMatching = 1
> ```

Thanks! I got this already in my `vimrc`. I'm thinking more broadly where the old behaviour was actually _less_ broken than the existing one hence the suggestion to revert.

**Comment by 2072** [comment]
OK, it behave better now but there are still some edge cases to fix:

What works 'as expected':
```php
/* standard arrow matching*/
$a->foo()
  ->bar(function () {
      $a = 1;
  });

$article->author = $this->Authors
                        ->find()
                        ->first();

Assert::that($vatRate)->equals($item->getVatRate())
                      ->true('The VAT rates differ');

/* arrow matching is skiping the () content */
Assert::that($vatRate)->equals(
    $item->getVatRate()
)
                      ->true('The VAT rates differ');


Assert::that($vatRate)->equals(
    $item->getVatRate())
                      ->true('The VAT rates differ');


/* no proper arrow to be matched resort to default last line + &sw indentation*/
Assert::that($vatRate->equals($item->getVatRate()))
    ->true('The VAT rates differ');

Assert::that($vatRate->equals(
    $item->getVatRate()))
    ->true('The VAT rates differ');

```
What doesn't work:
```php
/* bugged cases FIXME */

$article->author = $this->Authors( ytrt()->hhh()
    ->find() // resort to default indent bcause of () balance > 0 on last line
    ->first()
                           ); // untouched because phpindent returns <0 (0 - &sw) (legacy indent)


$article->author = $this->Authors(
    ytrt()->hhh()
          ->find()
          ->first()
  ); // bad indentation because no proper searchpair() call (legacy indent)
```

Time is difficult to find lately, I'll fix the remaining cases later. In the mean time if you could test this and find other edge cases, that would be great :)

**Comment by 2072** [comment]
The last 2 remaining cases were fixed by dffa2a500
